examples/EF_QAE/main.py

```diff
@@ -0,0 +1,265 @@
+                    circuit.add(gates.RY(q, theta=0))
+                    
+                cost = 0
+                circuit.set_parameters(params) # this will change all thetas to the appropriate values
+                for i in range(len(vector_0)):
+                    final_state = circuit.execute(np.copy(vector_0[i]))
+                    cost += encoder.expectation(final_state).numpy().real
+                for i in range(len(vector_1)):
+                    final_state = circuit.execute(np.copy(vector_1[i]))
+                    cost += encoder.expectation(final_state).numpy().real
+                    
+                cost_function_steps.append(cost/(len(vector_0)+len(vector_1))) # save cost function value after each step
+        
+                if count[0] % 50 == 0:
+                    print(count[0], cost/(len(vector_0)+len(vector_1)))
+                count[0] += 1
+        
+                return cost/(len(vector_0)+len(vector_1))
+        
+            nparams = 2 * nqubits * layers + compress
+            initial_params = np.random.uniform(0, 2*np.pi, nparams)
+        
+            result = minimize(cost_function_QAE_Digits, initial_params,
+                              args=(count), method='BFGS', options={'maxiter': 5.0e4})
+
+        if autoencoder == 0:
+            def cost_function_EF_QAE_Digits(params, count):
+                """Evaluates the cost function to be minimized for the EF-QAE and Handwritten digits.
+        
+                Args:
+                    params (array or list): values of the parameters.
+        
+                Returns:
+                    Value of the cost function.
+                """
+                cost = 0
+                for i in range(len(vector_0)):
+                    final_state = ansatz_EF_QAE(params, 1).execute(np.copy(vector_0[i]))
+                    cost += encoder.expectation(final_state).numpy().real
+                for i in range(len(vector_1)):
+                    final_state = ansatz_EF_QAE(params, 2).execute(np.copy(vector_1[i]))
+                    cost += encoder.expectation(final_state).numpy().real
+                    
+                cost_function_steps.append(cost/(len(vector_0)+len(vector_1))) # save cost function value after each step
+        
+                if count[0] % 50 == 0:
+                    print(count[0], cost/(len(vector_0)+len(vector_1)))
+                count[0] += 1
+        
+                return cost/(len(vector_0)+len(vector_1))
+        
+            
+            nparams = 4 * nqubits * layers + 2 * compress
+            initial_params = np.random.uniform(0, 2*np.pi, nparams)
+        
+            result = minimize(cost_function_EF_QAE_Digits, initial_params,
+                              args=(count), method='BFGS', options={'maxiter': 5.0e4})           
+            
+        else:
+            raise("You have to introduce a value of 0 or 1 in the autoencoder argument.")
```

Comment:
I get `TypeError: exceptions must derive from BaseException` from this line so I think the correct way to do it would be 
```Python
raise ValueError("You have to introduce a value of 0 or 1 in the autoencoder argument.")
```

Reply:
Thanks Stavros. I just fixed it.